internal/pkg/otel/manager/execution_subprocess.go

```diff
@@ -0,0 +1,210 @@
+	if cfg == nil {
+		// configuration is required
+		return nil, errors.New("no configuration provided")
+	}
+
+	if r.collectorPath == "" {
+		// collector path is required
+		return nil, errors.New("no collector path provided")
+	}
+
+	if _, err := os.Stat(r.collectorPath); err != nil {
+		// we cannot access the collector path
+		return nil, fmt.Errorf("cannot access collector path: %w", err)
+	}
+
+	httpHealthCheckPort, err := findRandomTCPPort()
+	if err != nil {
+		return nil, fmt.Errorf("could not find port for http health check: %w", err)
+	}
+
+	if err := injectHeathCheckV2Extension(cfg, httpHealthCheckPort); err != nil {
+		return nil, fmt.Errorf("failed to inject health check extension: %w", err)
+	}
+
+	confMap := cfg.ToStringMap()
+	confBytes, err := yaml.Marshal(confMap)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal config to yaml: %w", err)
+	}
+
+	stdOut := runtimeLogger.NewLogWriterWithDefaults(logger.Core(), zapcore.InfoLevel)
+	// info level for stdErr because by default collector writes to stderr
+	stdErr := runtimeLogger.NewLogWriterWithDefaults(logger.Core(), zapcore.InfoLevel)
+
+	procCtx, procCtxCancel := context.WithCancel(ctx)
+	processInfo, err := process.Start(r.collectorPath,
+		process.WithArgs(r.collectorArgs),
+		process.WithContext(procCtx),
+		process.WithEnv(os.Environ()),
+		process.WithCmdOptions(func(c *exec.Cmd) error {
+			c.Stdin = bytes.NewReader(confBytes)
+			c.Stdout = stdOut
+			c.Stderr = stdErr
+			return nil
+		}),
+	)
+	if err != nil {
+		// we failed to start the process
+		procCtxCancel()
+		return nil, fmt.Errorf("failed to start supervised collector: %w", err)
+	}
+	if processInfo.Process == nil {
+		// this should not happen but just in case
+		procCtxCancel()
+		return nil, fmt.Errorf("failed to start supervised collector: process is nil")
+	}
+
+	ctl := &procHandle{
+		processDoneCh: make(chan struct{}),
+		processInfo:   processInfo,
+	}
+
+	healthCheckDone := make(chan struct{})
+	go func() {
+		defer func() {
+			close(healthCheckDone)
+		}()
+		currentStatus := aggregateStatus(componentstatus.StatusStarting, nil)
+		reportStatus(ctx, statusCh, currentStatus)
+		// we will check the health of the collector every 1 second for the first 10 attempts
+		// until we get a successful response.
+		maxFailedAttempts := 10
+		timerDuration := 1 * time.Second
+		healthCheckTicker := time.NewTimer(timerDuration)
+		currentFailedAttempts := 0
+		defer healthCheckTicker.Stop()
+		for {
+			statuses, err := AllComponentsStatuses(procCtx, httpHealthCheckPort)
+			if err != nil {
+				switch {
+				case errors.Is(err, context.Canceled):
+					reportStatus(ctx, statusCh, aggregateStatus(componentstatus.StatusStopped, nil))
+					return
+				case currentFailedAttempts > maxFailedAttempts:
+					reportStatus(procCtx, statusCh, aggregateStatus(componentstatus.StatusFatalError, err))
```

Comment:
Technically this error could be transient and not fatal, we have an SDH right now where sub-process missed checkins due to cgroup CPU throttling on k8s.

Reply:
Switched to a `RecoverableError` here c3630f6edd5a8bc817232e86ddb4e031a8928aa8 and removed the giving up